### PR TITLE
Re #543, add Scenario class to test framework, add test for scope

### PR DESCRIPTION
**Pull request by sanjuroj**
- Add Scenario class which will carry methods for program level statistics
- Add scope statistic to ProgramPage context
- Add achieved-ratio getter for indicators in a Scenario
- Add a test for the three overall on-scope stats (low, high, and on-scope)
- Add multi-indicator scenario for testing stats
- Add ONSCOPE_MARGIN constant to Indicator model
- Put all remaining ScenarioBase tests into the IndicatorDetailMixin, in the hopes of providing better abstraction of the XBase classes.
- Created IndicatorStatsMixin to hold program level statistics tests

### CODE REVIEW

**Comment by sanjuroj** [comment]
Changed my mind, I think this will work for now.

File: tola/test/base_classes.py [hunk 1]
```diff
@@ -54,31 +56,72 @@ def url_name(self):
...
+        raise ImproperlyConfigured("A value for the url_name variable hasn't been set")
+
+    def setUp(self):
+        super(ScenarioBase2, self).setUp()
+        self.indicator.delete()
+        self.indicator_ids = instantiate_scenario(self.program.id, self.scenario)
+        self.indicators = Indicator.objects.filter(id__in=self.indicator_ids)
+        self.url = reverse_lazy(self.url_name, args=[self.program.id, 0, 0])
+        self.response = self.client.get(self.url)
+
```
Comment (kurtwall) [inline tola/test/base_classes.py]: Extra newline (PEP8)

File: tola/test/scenario_definitions.py [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 from indicators.models import Indicator
 
 indicator_scenarios = {}
 indicator_values = []
 
 """
+DON'T CHANGE THESE SCENARIOS UNLESS YOU'RE SURE YOU KNOW WHAT YOU'RE DOING!!!!
+These are meant to be common scenarios that can be used in many different tests.  If you change a scenario, 
+the change will ripple through all tests that use this scenario, potentially invalidating the tests that are based
+on it and creating a rift in space-time. If your test requires a very specific setup, you might be better off
```
Comment (kurtwall) [inline tola/test/scenario_definitions.py]: > creating a rift in space-time

Testing optimization strategy: open hole in space-time, see if tests _will_ pass or fail or in the future, then report _that_ in the present. Should reduce runtime to mere milliseconds.

Comment (sanjuroj) [reply tola/test/scenario_definitions.py]: Ah the [dangers](https://collections.slsa.sa.gov.au/resource/B+71885) of space-time holes.

**Pull request merged**